mona_sdk/authentication.py

```diff
@@ -121,29 +124,44 @@ def _get_error_string_from_token_info(api_key):
     return ", ".join(_get_token_info_by_api_key(api_key, ERRORS)) if error_list else ""
 
 
-def _request_access_token_with_retries(api_key, secret):
+def _request_access_token_with_retries(client):
```

Comment:
perhaps rename client to mona_client across the board - client is really too generic and in the future we might have other clients too